src/function/table/system/duckdb_functions.cpp

```diff
@@ -453,9 +497,101 @@ struct PragmaFunctionExtractor {
 	}
 };
 
+static vector<Value> ToValueVector(vector<string> &string_vector) {
+	vector<Value> result;
+	for (string &str : string_vector) {
+		result.emplace_back(Value(str));
+	}
+	return result;
+}
+
+template <class T, class OP>
+static Value GetParameterNames(FunctionEntry &entry, idx_t function_idx, FunctionDescription &function_description,
+                               Value &parameter_types) {
+	vector<Value> parameter_names;
+	if (!function_description.parameter_names.empty()) {
+		for (idx_t param_idx = 0; param_idx < ListValue::GetChildren(parameter_types).size(); param_idx++) {
+			if (param_idx < function_description.parameter_names.size()) {
+				parameter_names.emplace_back(function_description.parameter_names[param_idx]);
+			} else {
+				parameter_names.emplace_back("col" + to_string(param_idx));
+			}
+		}
+	} else {
+		// fallback
+		auto &function = entry.Cast<T>();
+		parameter_names = OP::GetParameters(function, function_idx);
+	}
+	return Value::LIST(LogicalType::VARCHAR, parameter_names);
+}
+
+// returns values:
+//  -1: no match; 0: explicit match; N: match by using N <ANY> values
+static int CalcDescriptionSpecificity(FunctionDescription &description, const vector<LogicalType> &parameter_types) {
+	int any_count = 0;
+
+	for (idx_t i = 0; i < description.parameter_types.size(); i++) {
+		if (description.parameter_types[i] != parameter_types[i] &&
+		    description.parameter_types[i].id() != LogicalTypeId::ANY) {
+			return -1;
+		}
+		if (description.parameter_types[i].id() == LogicalTypeId::ANY) {
+			any_count++;
+		}
+	}
+	return any_count;
+}
+
+/*
+Find FunctionDescription object with matching number of arguments and types
+return -1: no description matches the overload
+return >= 0: index of best matching description
+*/
+static int GetFunctionDescriptionIndex(vector<FunctionDescription> &function_descriptions,
```

Comment:
We have a `optional_idx` class you could use instead to make these sort of return values explicit